frontend/src/components/widgets/Form/FormClearHelper.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { isValidFormId } from "src/lib/utils"
+import { WidgetStateManager } from "src/lib/WidgetStateManager"
+import { SignalConnection } from "typed-signals"
+
+export class FormClearHelper {
```

Comment:
I think we should open an issue to translate our widgets to hooks, which will eliminate this logic.